docs/API.md

```diff
@@ -60,101 +71,170 @@ type Step = (page: Page, options?: Options) => void;
 
+<a name="signTypedData"></a>
+## `metaMask.signTypedData(): Promise<void>`  
+commands MetaMask to sign a message. For this to work MetaMask must be in a sign typed data confirmation state.
+
 <a name="approve"></a>
-## `metamask.approve(): Promise<void>`  
+## `metaMask.approve(): Promise<void>`  
 enables the app to connect to MetaMask account in privacy mode
 
 <a name="helpers"></a>
-## `metamask.helpers`
+# Helpers methods
 
 <a name="getTokenBalance"></a>
-### `metamask.helpers.getTokenBalance(tokenSymbol: string): Promise<number>`
+## `metaMask.helpers.getTokenBalance(tokenSymbol: string): Promise<number>`
 get balance of specific token
 
 <a name="deleteAccount"></a>
-### `metamask.helpers.deleteAccount(accountNumber: number): Promise<void>`
+## `metaMask.helpers.deleteAccount(accountNumber: number): Promise<void>`
 deletes account containing name with specified number
 
 <a name="deleteNetwork"></a>
-### `metamask.helpers.deleteNetwork(): Promise<void>`
-deletes custom network from metamask
+## `metaMask.helpers.deleteNetwork(): Promise<void>`
+deletes custom network from metaMask
 
 <a name="page"></a>
-## `metamask.page` is MetaMask plugin `Page`
+## `metaMask.page` is the MetaMask plugin `Page`
 **for advanced usages** in case you need custom features.
+
+<a name="snaps_methods"></a>
+# Snaps methods
+
+<a name="installSnap"></a>
+## `metaMask.snaps.installSnap: ( snapIdOrLocation: string, opts: { hasPermissions: boolean; hasKeyPermissions: boolean; customSteps?: InstallStep[]; version?: string;},installationSnapUrl?: string`
+```typescript
+/**
+  * Installs snap. Function will throw if there is an error while installing the snap.
+  * @param snapIdOrLocation either the snapId or the full path to your snap directory
+  * where we can find the bundled snap (you need to ensure the snap is built)
+  * @param opts {Object} the snap method you want to invoke
+  * @param opts.hasPermissions Set to true if the snap uses some permissions
+  * @param opts.hasKeyPermissions Set to true if the snap uses the key permissions
+  * @param installationSnapUrl the url of your dapp. Defaults to google.com
+  */
+) => Promise<string>;
+```
+
+<a name="invokeSnap"></a>
+## `metaMask.snaps.invokeSnap<Result = unknown, Params extends Serializable = Serializable>(page: DappeteerPage,snapId: string,method: string,params?: Params): Promise<Partial<Result>>`
+invoke a MetaMask snap method. Function will throw if there is an error while invoking snap.
+
+```typescript
+/**
+ * Use generic params to override result and parameter types.
+ * @param page Browser page where injected MetaMask provider will be available.
+ * For most snaps, openning google.com will suffice.
+ * @param snapId id of your installed snap (result of invoking `installSnap` method)
+ * @param method snap method you want to invoke
+ * @param params required parameters of snap method
+*/
+```
+
```

Comment:
I think this JSDoc is more useful than the other documentation we have in this page that is pretty minimal. Overall I think it would be nice if all the methods were as well documented as these ones, so that we could generate the documentation directly. I will take a look at this as part of #205 

In the mean time, I would leave this here since I don't think it can hurt.

Reply:
but here brokes overall style of the document

Reply:
Fair enough, I'll redo this part manually for now